src/cloudant/client.py

```diff
@@ -788,9 +788,17 @@ def bluemix(cls, vcap_services, instance_name=None, service_name=None, **kwargs)
             print client.all_dbs()
         """
         service_name = service_name or 'cloudantNoSQLDB'  # default service
-        service = CloudFoundryService(vcap_services,
-                                      instance_name=instance_name,
-                                      service_name=service_name)
+        try:
+            service = CloudFoundryService(vcap_services,
+                                          instance_name=instance_name,
+                                          service_name=service_name)
+        except CloudantException:
+            raise CloudantClientException(103)
+
+        if hasattr(service, 'iam_api_key'):
+            return Cloudant.iam(service.username,
+                                service.iam_api_key,
```

Comment:
How about `service.iam_api_key or service.password`?

Reply:
I originally used `service.iam_api_key`, but it will throw `AttributeError` if the attribute is not part of `CloudFoundryService`.  I have some tests where `iam_api_key` is not set to test the legacy creds logic.  
Also, it only seems that the updated VCAP version has IAM key or IAM key with legacy creds.  This would provide backward compatibility with VCAPs that have no IAM API key.